src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -29,12 +29,29 @@ public override async Task RunAsync(CancellationToken ct)
             EventProcessorClient.ProcessErrorAsync += ProcessErrorHandler;
             EventProcessorClient.PartitionInitializingAsync += ProcessInitializingHandler;
 
-            try
+            // Try to connect and read events
+            bool connected = false;
+            while (!connected)
             {
-                Console.WriteLine($"Starting event hub processor at {DateTime.UtcNow}");
-                await EventProcessorClient.StartProcessingAsync(ct);
+                try
+                {
+                    Logger.LogTrace($"Starting event hub processor at {DateTime.UtcNow}");
+                    await EventProcessorClient.StartProcessingAsync(ct);
+                    connected = true;
+                    break;
+                }
+                catch (AggregateException ex)
+                {
+                    HandleOwnershipFailure(ex);
+                }
+
+                Logger.LogTrace("Unable to read from event hub. Retrying in 1 minute");
+                await Task.Delay(TimeSpan.FromMinutes(1));
```

Comment:
We should pass in the `cancellationToken` into `Task.Delay`. In case a cancellation request is called on the token.